extensions/wikia/AdEngine/js/template/interstitial.js

```diff
@@ -0,0 +1,60 @@
+/*global define*/
+define('ext.wikia.adEngine.template.interstitial', [
+	'wikia.document',
+	'wikia.log',
+	'wikia.ui.factory',
+	'wikia.window'
+], function (doc, log, uiFactory, win) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.template.interstitial',
+		modalId = 'ext-wikia-adEngine-template-interstitial',
+		modalPadding = 20,
+		pollingInterval = 1000; // 1s
+
+	function show(params) {
+		log(['show', params], 'debug', logGroup);
+
+		var iframe = doc.getElementById(params.slot).querySelector('iframe'),
+			iframeBody = iframe.contentWindow.document.body,
+			selector = params.elem;
+
+		function onFound(element) {
+			var modalConfig = {
+					vars: {
+						id: modalId,
+						size: params.size || 'medium',
+						content: element.outerHTML,
+						title: 'Advertisement',
+						closeText: 'Close',
+						buttons: []
+					}
+				};
+
+			uiFactory.init('modal').then(function (uiModal) {
+				uiModal.createComponent(modalConfig, function (modal) {
+					var width = params.width || modal.$content[0].querySelector(selector).clientWidth;
+					modal.$element.width(width + 2 * modalPadding);
+					modal.show();
+				});
+			});
+		}
+
+		function waitForElement() {
```

Comment:
The element will be there in most of the cases, since the ad event fires after iframe's onload, so the polling is there only to support some edgy ads, where they load asynchronously.

I think the MutationObserver could come handy when updating the floating rail ad, so it reacts to stuff added to rail.